src/dmd/expression.d

```diff
@@ -1283,8 +1283,16 @@ extern (C++) abstract class Expression : ASTNode
             {
                 if (loc.linnum == 0) // e.g. implicitly generated dtor
                     loc = sc.func.loc;
-                error("`@nogc` %s `%s` cannot call non-@nogc %s `%s`",
-                    sc.func.kind(), sc.func.toPrettyChars(), f.kind(), f.toPrettyChars());
+
+                // Lowered non-@nogc'd hook will print their own error inside of nogc.d (NOGCVisitor.visit(CallExp e)).
+                // This stopped flag remove the unnessasary, more verbose, error message for the calling of the hook.
+                bool printError = true;
+                if (f.ident == Id._d_arraysetlengthT || f.ident == Id.HookTraceImpl)
+                    printError = false;
+
+                if (printError)
```

Comment:
Why not just `if (!(f.ident == Id._d_arraysetlengthT || f.ident == Id.HookTraceImpl))`?

Reply:
This is leftover after some debugging, will fix.